components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java

```diff
@@ -2106,4 +2282,176 @@ private void handleInternalTokenRevocation(String consumerKey, Properties proper
             }
         }
     }
+
+    /**
+     * FAPI validation to restrict the token endpoint authentication methods.
+     * Link - https://openid.net/specs/openid-financial-api-part-2-1_0.html#authorization-server (5.2.2 - 14)
+     * @param authenticationMethod authentication methid used to authenticate to the token endpoint
+     * @throws IdentityOAuthClientException
+     */
+    private void filterAllowedFAPITokenAuthMethods(String authenticationMethod) throws IdentityOAuthClientException {
+
+        List<String> allowedAuthMethods = IdentityUtil.getPropertyAsList(FAPI_CLIENT_AUTH_METHOD_CONFIGURATION);
+        if (allowedAuthMethods.isEmpty()) {
+            allowedAuthMethods.add(OAuthConstants.PRIVATE_KEY_JWT);
+            allowedAuthMethods.add(OAuthConstants.TLS_CLIENT_AUTH);
+        }
+        if (authenticationMethod != null && !allowedAuthMethods.contains(authenticationMethod)) {
+            throw handleClientError(INVALID_REQUEST, "Invalid token endpoint authentication method requested.");
+        }
+    }
+
+    /**
+     * FAPI validation to restrict the signature algorithms.
+     * Link - https://openid.net/specs/openid-financial-api-part-2-1_0.html#algorithm-considerations
+     * @param signatureAlgorithm signature algorithm used to sign the assertions.
+     * @throws IdentityOAuthClientException
+     */
+    private void filterAllowedFAPISignatureAlgorithms(String signatureAlgorithm)
+            throws IdentityOAuthClientException {
+
+        List<String> allowedSignatureAlgorithms = IdentityUtil
+                .getPropertyAsList(FAPI_SIGNATURE_ALGORITHM_CONFIGURATION);
+        if (allowedSignatureAlgorithms.isEmpty()) {
+            allowedSignatureAlgorithms.add(OAuthConstants.SignatureAlgorithms.ES256);
+            allowedSignatureAlgorithms.add(OAuthConstants.SignatureAlgorithms.PS256);
```

Comment:
Since default server configs contains these values no need to have a hardcoded value

Reply:
removed